src/nodejsWorker.ts

```diff
@@ -1,9 +1,46 @@
 var logPrefix = "LanguageWorkerConsoleLog";
+var ACTIVE_LTS_VERSION = "v8";
+var CURRENT_BRANCH_VERSION = "v10";
 var worker;
+
+// Try validating node version
+// NOTE: This method should be manually tested if changed as it is in a sensitive code path 
+//       and is JavaScript that runs on at least node version 0.10.28
+function validateNodeVersion(version) {
+    var message;

```

Comment:
We do not need E2E test. A unit test would be good enough to test the logic for version verification. Yes, you are right, that we do need a E2E test that sets the node version to something that is not supported. You can add that later.

Reply:
It’s really hard to unit test because nodejsWorker has javascript code that executes on being loaded with ‘require’ - so every mocha test that attempts to import nodejsWorker will either fail with missing reference ‘./Worker.js’ or will run worker.startWorker. I think the way to get around this is to separate the method into a separate file and add unit tests to that exported method, but that has the challenges I mentioned above with referencing and wepacking